src/Compilers/CSharp/Portable/Symbols/Symbol.cs

```diff
@@ -1100,6 +1100,31 @@ public ImmutableArray<SymbolDisplayPart> ToDisplayParts(SymbolDisplayFormat form
             }
         }
 
+        internal static void CheckForBlockAndExpressionBody(
+            CSharpSyntaxNode block,
+            CSharpSyntaxNode expression,
+            CSharpSyntaxNode syntax,
+            DiagnosticBag diagnostics)
+        {
+            if (block != null && expression != null)
+            {
+                ErrorCode code;
+                if (syntax is BaseMethodDeclarationSyntax ||
+                    syntax is AccessorDeclarationSyntax ||
+                    syntax is LocalFunctionStatementSyntax)
+                {
+                    code = ErrorCode.ERR_BlockBodyAndExpressionBody;
+                }
+                else
+                {
+                    Debug.Assert(syntax is BasePropertyDeclarationSyntax);
+                    code = ErrorCode.ERR_AccessorListAndExpressionBody;
+                }
```

Comment:
Alternatively, could we pass in the `ErrorCode` as a parameter to this method, defaulting to `ERR_BlockBodyAndExpressionBody`?